src/Apps/InfiniteDiscovery/InfiniteDiscoveryApp.tsx

```diff
@@ -0,0 +1,310 @@
+import { Flex, Button, Text, Separator } from "@artsy/palette"
+import { getMetaphysicsEndpoint } from "System/Relay/getMetaphysicsEndpoint"
+import React, { useEffect } from "react"
+import { uniq, sampleSize, shuffle } from "lodash"
+
+const ES_URL =
+  "https://vpc-artsy-staging-es7-srsqgcck5vzy7uq2eqqhvrm4x4.us-east-1.es.amazonaws.com"
+
+const BUILD_SEARCH_QUERY = (lArtworks, dArtworks) => {
+  const likedArtworks = lArtworks.map(artwork => {
+    return {
+      _index: "artworks_7shards_production",
+      _type: "_doc",
+      _id: artwork,
+    }
+  })
+
+  const dismissedArtworks = dArtworks.map(artwork => ({
+    term: {
+      _id: artwork,
+    },
+  }))
+
+  return {
+    query: {
+      bool: {
+        must: [
+          {
+            more_like_this: {
+              fields: ["id", "genes_vectors"],
+              like: likedArtworks,
+              min_term_freq: 1,
+              max_query_terms: 12,
+            },
+          },
+        ],
+        filter: [
+          {
+            term: { published: true },
+          },
+          {
+            term: { for_sale: true },
+          },
+          { range: { career_stage: { gte: 0, lte: 100 } } },
+        ],
+        must_not: dismissedArtworks,
+      },
+    },
+    aggregations: {
+      uniq_artists: {
+        terms: {
+          field: "artist_id",
+          size: 1,
+          order: {
+            top_score: "desc",
+          },
+        },
+        aggs: {
+          sim_works: {
+            top_hits: {
+              size: 1,
+              _source: ["_id"],
+            },
+          },
+          top_score: {
+            max: {
+              script: "_score",
+            },
+          },
+        },
+      },
+    },
+  }
+}
```

Comment:
Possible to get a quick rundown of the underlying logic returned back from this query? Helpful for those with less ES expertise.